packages/components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -42,11 +42,20 @@ export class TabNav {
   handleClick(event: MouseEvent) {
     this.removeFirstRenderAttr();
 
-    const nextTab = event.target as HTMLScaleTabHeaderElement;
-    if (nextTab.getAttribute('role') !== 'tab') {
-      return;
+    // workaround for slotted icons
+    const targetHTMLElement = event.target as HTMLElement;
+    const targetTag = targetHTMLElement.tagName.toLowerCase();
+    const svgTags = ['svg', 'g', 'path'];
+
+    if (svgTags.includes(targetTag)) {
+      const closestNextTab = targetHTMLElement.closest(
+        `scale-tab-header[role="tab"]`
+      ) as HTMLScaleTabHeaderElement;
+      closestNextTab ? this.selectTab(closestNextTab) : null;
+    } else {
+      const nextTab = event.target as HTMLScaleTabHeaderElement;
+      nextTab.getAttribute('role') === 'tab' ? this.selectTab(nextTab) : null;
     }
-    this.selectTab(nextTab);
```

Comment:
Maybe you can use a `let` for `nextTab` defined before the if/else statement and keep this `this.selectTab(nextTab)` call at the end?